tsdb/index.go

```diff
@@ -2454,7 +2454,7 @@ func (is IndexSet) MeasurementSeriesKeysByExpr(name []byte, expr influxql.Expr)
 		}
 
 		name, tags := ParseSeriesKey(seriesKey)
-		if name == nil && tags == nil {
+		if name == nil || tags == nil {
```

Comment:
Are you sure we don't ever want to add a key with no tags here?  I'm not sure, I just wanted you to verify exactly what was the right test. exactly what